pkg/webui/console/components/collaborator-form/index.js

```diff
@@ -0,0 +1,241 @@
+
+    try {
+      await onSubmit(collaborator)
+      resetForm(values)
+      onSubmitSuccess()
+    } catch (error) {
+      setSubmitting(false)
+      this.setState({ error })
+      onSubmitFailure(error)
+    }
+  }
+
+  async handleDelete () {
+    const { collaborator, onDelete, onDeleteSuccess } = this.props
+    const collaborator_type = collaborator.isUser ? 'user' : 'organization'
+
+    const collaborator_ids = {
+      [`${collaborator_type}_ids`]: {
+        [`${collaborator_type}_id`]: collaborator.id,
+      },
+    }
+    const updatedCollaborator = {
+      ids: collaborator_ids,
+    }
+
+    try {
+      await onDelete(updatedCollaborator)
+      toast({
+        message: sharedMessages.collaboratorDeleteSuccess,
+        type: toast.types.SUCCESS,
+      })
+      onDeleteSuccess()
+    } catch (error) {
+      await this.setState({ error })
+    }
+  }
+
+  computeInitialValues () {
+    const { collaborator } = this.props
+
+    if (!collaborator) {
+      return {
+        collaborator_id: '',
+        collaborator_type: 'user',
+        rights: [],
+      }
+    }
+
+    return {
+      collaborator_id: collaborator.id,
+      collaborator_type: collaborator.isUser ? 'user' : 'organization',
+      rights: [ ...collaborator.rights ],
+    }
+  }
+
+  render () {
+    const {
+      collaborator,
+      rights,
+      universalRights,
+      error: passedError,
+      update,
+    } = this.props
+
+    const { error: submitError } = this.state
+
+    const error = passedError || submitError
+
+    const hasRightAll = Boolean(collaborator && collaborator.rights.includes(RIGHT_ALL))
+
+    return (
+      <Form
+        horizontal
+        error={error}
+        onSubmit={this.handleSubmit}
+        initialValues={this.computeInitialValues()}
+        validationSchema={validationSchema}
+      >
+        <Message
+          component="h4"
+          content={sharedMessages.generalInformation}
+        />
+        <Form.Field
+          name="collaborator_id"
+          component={Input}
+          title={sharedMessages.collaboratorId}
+          required
+          autoFocus={!update}
+          disabled={update}
+        />
+        <Form.Field
+          name="collaborator_type"
+          title={sharedMessages.type}
+          component={Radio.Group}
+          horizontal={false}
+          disabled={update}
+          required
+        >
+          <Radio
+            label={sharedMessages.user}
+            value="user"
+          />
+          <Radio
+            label={sharedMessages.organization}
+            value="organization"
+          />
+        </Form.Field>
+        { hasRightAll && <Notification small info={m.cannotModifyRightAll} />}
+        <Form.Field
+          name="rights"
+          title={sharedMessages.rights}
+          required
+          component={RightsGroup}
+          rights={rights}
+          universalRight={universalRights.length ? universalRights[0] : undefined}
```

Comment:
`universalRights` should default to an empty array and you can simply pass `universalRights[0]`